File: numberTheory/2247.py
def counting_star(N):
    matrix = []
    for i in range(3 * len(N)):
        if i // len(N) == 1:
            matrix.append(N[i % len(N)] + " " * len(N) + N[i % len(N)])
        else:
            matrix.append(N[i % len(N)] * 3)
    return list(matrix)


star = ["***", "* *", "***"]
N = int(input())
k = 0

while N != 3:
    N = int(N / 3)
    k += 1

for i in range(k):
    star = counting_star(star)

for i in star:
    print(i)

File: numberTheory/2247.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");
const N = parseInt(input[0]);

function counting_star(N) {
  const matrix = [];
  for (let i = 0; i < 3 * N.length; i++) {
    if (Math.floor(i / N.length) === 1) {
      matrix.push(N[i % N.length] + " ".repeat(N.length) + N[i % N.length]);
    } else {
      matrix.push(N[i % N.length].repeat(3));
    }
  }
  return matrix;
}

let star = ["***", "* *", "***"];
let k = 0;

while (N !== 3) {
  N = Math.floor(N / 3);
  k += 1;
}

for (let i = 0; i < k; i++) {
  star = counting_star(star);
}

for (let i = 0; i < star.length; i++) {
  console.log(star[i]);
}
